jetty-ee10/jetty-ee10-websocket/jetty-ee10-websocket-jakarta-common/src/test/java/org/eclipse/jetty/ee10/websocket/jakarta/common/AbstractJakartaWebSocketFrameHandlerTest.java

```diff
@@ -22,25 +22,30 @@
 import org.eclipse.jetty.ee10.websocket.jakarta.common.encoders.AvailableEncoders;
 import org.eclipse.jetty.websocket.core.CoreSession;
 import org.eclipse.jetty.websocket.core.WebSocketComponents;
-import org.junit.jupiter.api.AfterAll;
-import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
 
 public abstract class AbstractJakartaWebSocketFrameHandlerTest
 {
-    protected static DummyContainer container;
-    private static WebSocketComponents components;
+    protected DummyContainer container;
+    private WebSocketComponents components;
 
-    @BeforeAll
-    public static void initContainer() throws Exception
+    @BeforeEach
+    public void startContainer() throws Exception
     {
         container = new DummyContainer();
         container.start();
         components = new WebSocketComponents();
         components.start();
+
+        endpointConfig = ClientEndpointConfig.Builder.create().build();
+        encoders = new AvailableEncoders(endpointConfig, coreSession.getWebSocketComponents());
+        decoders = new AvailableDecoders(endpointConfig, coreSession.getWebSocketComponents());
+        uriParams = new HashMap<>();
     }
 
-    @AfterAll
-    public static void stopContainer() throws Exception
+    @AfterEach
+    public void stopContainer() throws Exception
     {
         components.stop();
```

Comment:
You should really use `LifeCycle.stop(components)` and `LifeCycle.stop(container)`.

If, for some reason, in `startContainer()` an exception is thrown, the field will be `null`, and calling `stop()` will cause an NPE.